File: solutions/python/largest-series-product/1/largest_series_product.py
def largest_product(series, size):
    if size < 0:
        raise ValueError('span must not be negative')
    if size > len(series):
        raise ValueError('span must be smaller than string length')
    if series == '' or size == 0:
        return 1

    if not series.isdigit():
        raise ValueError('digits input must only contain digits')

    numbers = [int(char) for char in series]

    max_product = 0
    for i in range(len(numbers) - size + 1):
        product = 1
        for j in range(i, i + size):
            product *= numbers[j]
        max_product = max(max_product, product)

    return max_product

File: solutions/typescript/largest-series-product/1/largest-series-product.ts
export const largestProduct = (input: string, count: number): number => {
  if ( count < 0 ) throw('Span must be greater than zero');
  if ( count > input.length ) throw('Span must be smaller than string length');
  if ( input === '' || count === 0 ) return 1;

  const numbers = input.split('')
    .map((n: string): number => parseInt(n, 10))
    .filter((n: number): boolean => {
      if ( Number.isNaN(n) ) throw('Digits input must only contain digits');
      return true;
    });

  let max = 0;
  for ( let i = 0; i <= numbers.length - count; i++) {
    let sum = numbers
      .slice(i, i + count)
      .reduce((acc: number, cur: number): number => acc * cur);
    if ( sum > max ) max = sum;
  }
  return max;
}
